community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/BaseRecordFormats.java

```diff
@@ -0,0 +1,124 @@
+ */
+package org.neo4j.kernel.impl.store.format;
+
+import org.neo4j.kernel.impl.store.StoreType;
+
+import static org.neo4j.helpers.ArrayUtil.contains;
+import static org.neo4j.helpers.ArrayUtil.containsAll;
+import static org.neo4j.helpers.ArrayUtil.filter;
+
+/**
+ * Base class for simpler implementation of {@link RecordFormats}.
+ */
+public abstract class BaseRecordFormats implements RecordFormats
+{
+    private final int generation;
+    private final Capability[] capabilities;
+    private final String storeVersion;
+
+    protected BaseRecordFormats( String storeVersion, int generation, Capability... capabilities )
+    {
+        this.storeVersion = storeVersion;
+        this.generation = generation;
+        this.capabilities = capabilities;
+    }
+
+    @Override
+    public String storeVersion()
+    {
+        return storeVersion;
+    }
+
+    @Override
+    public boolean equals( Object obj )
+    {
+        if ( obj == null || !(obj instanceof RecordFormats) )
+        {
+            return false;
+        }
+
+        RecordFormats other = (RecordFormats) obj;
+        return  node().equals( other.node() ) &&
+                relationship().equals( other.relationship() ) &&
+                relationshipGroup().equals( other.relationshipGroup() ) &&
+                property().equals( other.property() ) &&
+                labelToken().equals( other.labelToken() ) &&
+                relationshipTypeToken().equals( other.relationshipTypeToken() ) &&
+                propertyKeyToken().equals( other.propertyKeyToken() ) &&
+                dynamic().equals( other.dynamic() );
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int hashCode = 17;
+        hashCode = 31 * hashCode + node().hashCode();
+        hashCode = 31 * hashCode + relationship().hashCode();
+        hashCode = 31 * hashCode + relationshipGroup().hashCode();
+        hashCode = 31 * hashCode + property().hashCode();
+        hashCode = 31 * hashCode + labelToken().hashCode();
+        hashCode = 31 * hashCode + relationshipTypeToken().hashCode();
+        hashCode = 31 * hashCode + propertyKeyToken().hashCode();
+        hashCode = 31 * hashCode + dynamic().hashCode();
+        return hashCode;
+    }
+
+    @Override
+    public String toString()
+    {
+        return "RecordFormat:" + getClass().getSimpleName() + "[" + storeVersion() + "]";
+    }
+
+    @Override
+    public int generation()
+    {
+        return generation;
+    }
+
+    @Override
+    public boolean hasStore( StoreType store )
+    {
+        return true;
+    }
+
+    @Override
+    public Capability[] capabilities()
+    {
+        return capabilities;
+    }
+
+    @Override
+    public boolean hasCapability( Capability capability )
+    {
+        return contains( capabilities(), capability );
+    }
+
+    @Override
+    public boolean hasSameCapabilities( RecordFormats other, int types )
+    {
+        Capability[] myFormatCapabilities =
```

Comment:
We don't want to be dependent on items in the array being in the exact same order though.